src/js/http_outgoing.js

```diff
@@ -152,8 +152,8 @@ OutgoingMessage.prototype._storeHeader = function(statusLine) {
 
 OutgoingMessage.prototype.setHeader = function(name, value) {
 
-  if (this._headers === null) {
-    this._headers = {};
+  if (util.isNull(this._headers)) {
+    this._headers = [];
```

Comment:
You are right! I will separate the coverage and fix in the future. The other hand, you are right again, there is no point to use array instead of object, this was my inattention.